### PR TITLE
Fixes data races for pkg/kubelet/config/file_linux_test.go

### LINKED ISSUE #64655: Data race in pkg/kubelet/config.watchFileChanged
**What happened**:
Unit test failed with a data race in a seemingly unrelated PR https://github.com/kubernetes/kubernetes/pull/63284
Logs: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/63284/pull-kubernetes-bazel-test/47917/
```
==================
WARNING: DATA RACE
Write at 0x00c42051c3c0 by goroutine 97:
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged.func1.3()
      pkg/kubelet/config/file_linux_test.go:341 +0x26a

Previous read at 0x00c42051c3c0 by goroutine 106:
  reflect.typedmemmove()
      GOROOT/src/runtime/mbarrier.go:265 +0x0
  reflect.packEface()
      GOROOT/src/reflect/value.go:119 +0x101
  reflect.valueInterface()
      GOROOT/src/reflect/value.go:978 +0x16a
  reflect.Value.Interface()
      GOROOT/src/reflect/value.go:948 +0x51
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:221 +0x517
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186 +0x9da
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:172 +0xd7e
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186 +0x9da
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186 +0x9da
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:183 +0x14d7
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:172 +0xd7e
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186 +0x9da
  k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.DeepEqual()
      vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:243 +0x2d5
  k8s.io/kubernetes/pkg/kubelet/config.expectUpdate()
      pkg/kubelet/config/file_linux_test.go:400 +0x669
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged.func1()
      pkg/kubelet/config/file_linux_test.go:352 +0x7b0
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged()
      pkg/kubelet/config/file_linux_test.go:363 +0xe2
  k8s.io/kubernetes/pkg/kubelet/config.TestWatchFileChanged()
      pkg/kubelet/config/file_linux_test.go:129 +0x96
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Goroutine 97 (running) created at:
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged.func1()
      pkg/kubelet/config/file_linux_test.go:350 +0x771
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged()
      pkg/kubelet/config/file_linux_test.go:363 +0xe2
  k8s.io/kubernetes/pkg/kubelet/config.TestWatchFileChanged()
      pkg/kubelet/config/file_linux_test.go:129 +0x96
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Goroutine 106 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/kubelet/config/linux_amd64_race_stripped/go_default_test~/testmain.go:179 +0x3ad
==================
==================
WARNING: DATA RACE
Read at 0x00c420726aa0 by goroutine 97:
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged.func1.3()
      pkg/kubelet/config/file_linux_test.go:343 +0x2e7

Previous write at 0x00c420726aa0 by goroutine 106:
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged.func1()
      pkg/kubelet/config/file_linux_test.go:356 +0x80d
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged()
      pkg/kubelet/config/file_linux_test.go:363 +0xe2
  k8s.io/kubernetes/pkg/kubelet/config.TestWatchFileChanged()
      pkg/kubelet/config/file_linux_test.go:129 +0x96
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Goroutine 97 (running) created at:
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged.func1()
      pkg/kubelet/config/file_linux_test.go:350 +0x771
  k8s.io/kubernetes/pkg/kubelet/config.watchFileChanged()
      pkg/kubelet/config/file_linux_test.go:363 +0xe2
  k8s.io/kubernetes/pkg/kubelet/config.TestWatchFileChanged()
      pkg/kubelet/config/file_linux_test.go:129 +0x96
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Goroutine 106 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/kubelet/config/linux_amd64_race_stripped/go_default_test~/testmain.go:179 +0x3ad
==================
E0602 12:54:50.094795      15 file_linux.go:61] Unable to read config path "/tmp/dir-test733403562": error while processing inotify event ("/tmp/dir-test733403562": 0x400 == IN_DELETE_SELF): the watched path is deleted
--- FAIL: TestWatchFileChanged (0.88s)
	file_linux_test.go:417: expected empty update, got types.PodUpdate{Pods:[]*v1.Pod{(*v1.Pod)(0xc420575c00)}, Op:0, Source:"file"}
	file_linux_test.go:372: unable to remove test file /tmp/dir-test733403562/test_pod_manifest_0_ch: remove /tmp/dir-test733403562/test_pod_manifest_0_ch: no such file or directory
	testing.go:730: race detected during execution of test
FAIL
```

I couldn't find an existing report about the issue so here it is.

/kind bug
/sig cli

### PR DESCRIPTION
**Pull request by yue9944882**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #64655

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yue9944882** [comment]
@kubernetes/sig-node-pr-reviews Friendly ping for a review :)

**Comment by mtaufen** [comment]
Thanks! Taking a look now.

File: pkg/kubelet/config/file_linux_test.go [hunk 1]
```diff
@@ -365,8 +364,12 @@ func watchFileChanged(watchDir bool, symlink bool, t *testing.T) {
 }
 
 func deleteFile(dir, file string, ch chan interface{}, t *testing.T) {
+	path := filepath.Join(dir, file)
+	if _, err := os.Stat(path); err != nil {
+		// The file might absent if it's renamed, deleted..
```
Comment (mtaufen) [inline pkg/kubelet/config/file_linux_test.go]: If it's been renamed, we still have to clean up the file at the new location, right?

**Review by mtaufen — COMMENTED** [review]
IMO this whole file could be refactored, it's rather hard to follow as written.

But I think the new locks are where I'd expect to see them, given the stack trace in #64655, so as a bug fix this looks ok to me for now.

I have one question regarding cleaning up files renamed during a test.

**Comment by mtaufen** [comment]
If you saw 0 garbage after running the test, that sounds good to me.
/lgtm
/assign @Random-Liu 
for approval
